moped-editor/src/views/projects/projectsListView/helpers.js

```diff
@@ -137,10 +143,35 @@ export const useTableComponents = ({
 /**
  * The Material Table column settings
  */
-export const useColumns = ({ hiddenColumns, classes }) => {
+export const useColumns = () => {
   const location = useLocation();
   const queryString = location.search;
 
+  const [hiddenColumns, setHiddenColumns] = useState(
+    JSON.parse(localStorage.getItem("mopedColumnConfig")) ?? DEFAULT_HIDDEN_COLS
+  );
+
+  const COLUMN_CONFIG = PROJECT_LIST_VIEW_QUERY_CONFIG.columns;
+
+  /*
+   * Sync hidden columns state with local storage
+   */
+  useEffect(() => {
+    localStorage.setItem("mopedColumnConfig", JSON.stringify(hiddenColumns));
+  }, [hiddenColumns]);
```

Comment:
I totally agree on this because I left this code thinking - how would I know that the local storage sync was even happening? Adding a separate custom hook at the top level of `ProjectListViewTable` sounds like the answer to me. Thanks!

Reply:
thanks for revisiting this—it looks good!